src/System.Management.Automation/engine/Interop/Windows/CertEnumSystemStore.cs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    [SuppressMessage("StyleCop.CSharp.NamingRules", "SA1305:FieldNamesMustNotUseHungarianNotation", Justification = "Keep native meth arg names.")]
+    internal static unsafe partial class Windows
+    {
+        [LibraryImport("Crypt32.dll", EntryPoint = "CertEnumSystemStore", SetLastError = true, StringMarshalling = StringMarshalling.Utf16)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        internal static unsafe partial bool CertEnumSystemStore(
+            uint dwFlags,
+            [MarshalAs(UnmanagedType.LPWStr)] string? pvSystemStoreLocation,
+            nint pvArg,
+            PfnCertEnumSystemStore pfnEnum);
+
+        internal delegate bool PfnCertEnumSystemStore(
+            [MarshalAs(UnmanagedType.LPWStr)] string storeName,
+            uint dwFlags,
+            nint pStoreInfo,
+            nint pvReserved,
+            nint pvArg);
```

Comment:
> This one I'm not sure about...it's been a while since I wrote P/Invoke code. Since it's a delegate, does it need the `MarshalAs` attribute on the `string`?

Yeah, same rules apply.

> Note that it's missing from the `bool`. I would suppose it's a both or neither situation.

The `MarshalAs` for bool is just good practice. The default behavior for the marshaller is to assume you mean a win32 `BOOL` (int32) rather than a managed bool (byte)